Subject: Patching detected hardening problems in libaria.
Author: Srećko Jurić-Kavelj <srecko.juric-kavelj@fer.hr>
--- a/src/ArMap.cpp
+++ b/src/ArMap.cpp
@@ -1605,7 +1605,7 @@
 {
   myConfigMapName[0] = '\0';
   if (fileName != NULL) {
-    snprintf(myConfigMapName, MAX_MAP_NAME_LENGTH, fileName);
+    snprintf(myConfigMapName, MAX_MAP_NAME_LENGTH, "%s", fileName);
   }
 
 } // end method changeConfigMapName
--- a/src/ArMapInterface.cpp
+++ b/src/ArMapInterface.cpp
@@ -101,7 +101,7 @@
     char *nameBuf = new char[totalLen];
     nameBuf[0] = '\0';
 
-    snprintf(nameBuf, totalLen, baseDirectory);
+    snprintf(nameBuf, totalLen, "%s", baseDirectory);
     ArUtil::appendSlash(nameBuf, totalLen);
     
     realFileName = nameBuf;
--- a/src/ArSocket.cpp
+++ b/src/ArSocket.cpp
@@ -527,7 +527,7 @@
   }
   if (separator.getArgc() == 1)
   {
-    snprintf(useHost, useHostSize, separator.getArg(0));
+    snprintf(useHost, useHostSize, "%s", separator.getArg(0));
     *port = rawPort;
     return;
   }
@@ -535,7 +535,7 @@
   {
     if (separator.isArgInt(1))
     {
-      snprintf(useHost, useHostSize, separator.getArg(0));
+      snprintf(useHost, useHostSize, "%s", separator.getArg(0));
       *port = separator.getArgInt(1);
       return;
     }
--- a/src/ArSocket_LIN.cpp
+++ b/src/ArSocket_LIN.cpp
@@ -678,7 +678,7 @@
   {
     myErrorStr="Trying to get socket name on an unopened socket";
     myError = NetFail;
-    printf(myErrorStr.c_str());
+    printf("%s", myErrorStr.c_str());
     return(false);
   }
 
--- a/Makefile
+++ b/Makefile
@@ -24,10 +24,10 @@
 # this is set up with the extra layer since the python wrapper needs exceptions
 # but I didn't want to have two sets of defines
 BARECXXFLAGS:=-g -Wall -D_REENTRANT  #-pg -fprofile-arcs
-CXXFLAGS+=$(BARECXXFLAGS) -fno-exceptions 
+CXXFLAGS+=$(CPPFLAGS) $(BARECXXFLAGS) -fno-exceptions 
 CXXINC:=-Iinclude
-CXXLINK=-Llib -lAria
-CXXSTATICLINK:=-Llib -Xlinker -Bstatic -lAria -Xlinker -Bdynamic 
+CXXLINK=$(LDFLAGS) -Llib -lAria
+CXXSTATICLINK:=$(LDFLAGS) -Llib -Xlinker -Bstatic -lAria -Xlinker -Bdynamic 
 
 host:=$(shell uname | cut -d _ -f 1)
 ifeq ($(host),MINGW32)
@@ -589,7 +589,7 @@
 ####
 
 lib/libAria.so: $(OFILES) Makefile.dep 
-	$(CXX) -shared -Wl,-soname,libAria.so.2 -o $(@).2.8.0 $(OFILES)
+	$(CXX) -shared -Wl,-soname,libAria.so.2 $(LDFLAGS) -o $(@).2.8.0 $(OFILES)
 	ln -s libAria.so.2.8.0 lib/libAria.so.2
 	ln -s libAria.so.2 lib/libAria.so
 
