--- a/Makefile
+++ b/Makefile
@@ -230,7 +230,7 @@
 ####
 
 # Default Rule
-all: dirs $(TARGETS) params docs python python-doc
+all: dirs $(TARGETS) params docs python python-doc java
 
 # Build all targets, docs, params, etc. etc.
 everything: dirs $(ALL_TARGETS) 
@@ -503,7 +503,7 @@
 # It is much faster to compile them all at once. Use ArRobot.java/class as a
 # stand-in for all java files and all class files.
 java/com/mobilerobots/Aria/ArRobot.class: java/com/mobilerobots/Aria/ArRobot.java
-		rm java/com/mobilerobots/Aria/*.class; $(JAVAC) -classpath java java/com/mobilerobots/Aria/*.java 
+	$(JAVAC) -classpath java java/com/mobilerobots/Aria/*.java
 
 
 java/ArNetworking.jar: FORCE
