Description: Building Python package from SWIG generated wrapper.
Author: Srećko Jurić-Kavelj <srecko.juric-kavelj@fer.hr>
--- a/Makefile
+++ b/Makefile
@@ -230,7 +230,7 @@
 ####
 
 # Default Rule
-all: dirs $(TARGETS) params docs
+all: dirs $(TARGETS) params docs python python-doc
 
 # Build all targets, docs, params, etc. etc.
 everything: dirs $(ALL_TARGETS) 
@@ -440,15 +440,6 @@
 
 ### Python wrapper: ###
 
-
-ifdef PYTHON_INCLUDE
-PYTHON_INCLUDE_FLAGS=-I$(PYTHON_INCLUDE)
-else
-$(warning WARNING using default value of /usr/include/python2.5 for PYTHON_INCLUDE directory. Set PYTHON_INCLUDE environment variable if you would like to use a different version of the Python C development library to build the Python ARIA wrapper.)
-PYTHON_INCLUDE:=/usr/include/python2.5
-PYTHON_INCLUDE_FLAGS=-I/usr/include/python2.5
-endif
-
 python: python/_AriaPy.so python/AriaPy.py
 
 py: python
@@ -456,21 +447,18 @@
 python-doc: python/AriaPy.html
 
 python/AriaPy.html: python/AriaPy.py
-	cd python; pydoc -w AriaPy
+	cd python; LD_LIBRARY_PATH=../lib pydoc -w AriaPy
 
 python/AriaPy_wrap.cpp python/AriaPy.py: include/wrapper.i include/*.h
 	-rm -f `find python -maxdepth 1 -xtype f -name "*Aria*" | grep -v .ds | grep -v .sln | grep -v .vcproj`
 	cd python; $(SWIG) -Wall -c++ -python -modern -module AriaPy -Dlinux -DAREXPORT -o AriaPy_wrap.cpp -I../include ../include/wrapper.i 
 
 python/_AriaPy.so: obj/AriaPy_wrap.o lib/libAria.so Makefile.dep
-	if ! test -f $(PYTHON_INCLUDE)/Python.h; then echo Error: $(PYTHON_INCLUDE)/Python.h not found. Is the Python development package installed on this system?; exit 1; fi
-	$(CXX) -shared -o $(@) obj/AriaPy_wrap.o -lpthread -ldl -lrt -Llib -lAria
+	$(CXX) -shared -o $(@) obj/AriaPy_wrap.o $(LDFLAGS) -Llib -lAria -lpthread -ldl -lrt
 
 obj/AriaPy_wrap.o: python/AriaPy_wrap.cpp
 	mkdir -p obj
-	@ if test -z "$(PYTHON_INCLUDE)"; then echo "*** Error: PYTHON_INCLUDE is not set, cannot build ARIA python wrapper"; fi
-	if ! test -f $(PYTHON_INCLUDE)/Python.h; then echo Error: $(PYTHON_INCLUDE)/Python.h not found. Is the Python development package installed on this system?; exit 1; fi
-	$(CXX) -c $(BARECXXFLAGS) $(CXXINC) $(PYTHON_INCLUDE_FLAGS) $< -o $@
+	$(CXX) -c $(CPPFLAGS) -O2 $(BARECXXFLAGS) $(CXXINC) $(shell python-config --includes) $< -o $@
 
 
 ### Java Wrapper: ###
@@ -668,6 +656,9 @@
 		if test \! -d $(DESTDIR)$(SYSTEM_ETC_DIR); then install -d $(DESTDIR)$(SYSTEM_ETC_DIR); fi ;\
 		echo $(STORED_INSTALL_DIR) > $(DESTDIR)$(SYSTEM_ETC_DIR)/Aria ;\
 	fi
+	$(INSTALL) -m 755 -d $(DESTDIR)$(INSTALL_DIR)/lib/$(shell pyversions -d)/dist-packages/
+	$(INSTALL) -D -m 644 python/_AriaPy.so $(DESTDIR)$(INSTALL_DIR)/lib/$(shell pyversions -d)/dist-packages/
+	$(INSTALL) -D -m 644 python/AriaPy.py $(DESTDIR)$(INSTALL_DIR)/lib/$(shell pyversions -d)/dist-packages/
 
 #		echo ;\
 #		echo       The files are owned by the group \"users\", and all members of that group ;\
